README.md

```diff
@@ -42,144 +76,321 @@ The only compile time dependency you will need is `okta-sdk-api`.  You will also
+* An [API token](https://developer.okta.com/docs/api/getting_started/getting_a_token)
+ 
+Construct a client instance by passing it your Okta domain name and API token:
+ 
+[//]: # (NOTE: code snippets in this README are updated automatically via a Maven plugin by running: mvn okta-code-snippet:snip.okta:okta-code-snippet-maven-plugin:snip)
+ 
+[//]: # (method: createClient)
+```java
+Client client = Clients.builder()
+    .setOrgUrl("https://dev-123456.oktapreview.com/")
+    .setClientCredentials(new TokenClientCredentials("your-api-token"))
+    .build();
+```
+[//]: # (end: createClient)
+ 
+Hard-coding the Okta domain and API token works for quick tests, but for real projects you should use a more secure way of storing these values (such as environment variables). This library supports a few different configuration sources, covered in the [configuration reference](#configuration-reference) section.
+ 
+## Usage guide
 
-``` java
-// Instantiate a builder for your Client. If needed, settings like Proxy and Caching can be defined here.
-ClientBuilder builder = Clients.builder();
+These examples will help you understand how to use this library. You can also browse the full [API reference documentation][javadocs].
 
-// No need to define anything else; build the Client instance. The ClientCredential information will be automatically found
-// in pre-defined locations.
-Client client = builder.build();
-```
+Once you initialize a `Client`, you can call methods to make requests to the Okta API.
 
-For more details see: [Creating a Client](https://github.com/okta/okta-sdk-java/wiki/Creating-a-Client)
+### Authenticate a User
 
-## Client CRUD Operations
+This library should be used with the Okta management API. For authentication, we recommend using an OAuth2 or OIDC library such as [Spring Security OAuth](https://spring.io/projects/spring-security-oauth) or [Okta's Spring Boot integration](https://github.com/okta/okta-spring-boot).
 
-The client is used to perform CRUD operations against Okta's management APIs. 
+### Get a User
 
-Create a group:
+[//]: # (method: getUser)
 ```java
+User user = client.getUser("a-user-id");
+```
+[//]: # (end: getUser)
 
-UserGroup group = GroupBuilder.instance()
-        .setName("my-user-group-" + UUID.randomUUID().toString())
-        .setDescription("Quickstart created Group")
-        .buildAndCreate(client);
+### List all Users
 
-// print a couple of the attributes
-println("Group: '" + group.getId() + "' was last updated on: " + group.getLastUpdated());
+[//]: # (method: listAllUsers)
+```java
+UserList users = client.listUsers();
 
+// stream
+client.listUsers().stream()
+    .forEach(user -> {
+      // do something
+    });
 ```
+[//]: # (end: listAllUsers)
 
+### Filter or search for Users
 
-Create a User Account:
+[//]: # (method: userSearch)
+```java
+UserList users = client.listUsers("query-string", null, null, null, null);
```

Comment:
What are the `null`s? Would I ever pass something there?

Reply:
Also, what is an example of a `query-string` I can create?

Reply:
The null's are just unused query params